test/CMakeLists.txt

```diff
@@ -1,4 +1,9 @@
 add_executable(unittests unittests.cpp)
 target_link_libraries(unittests bls12-381)
 
-add_test(NAME bls12-381 COMMAND unittests)
\ No newline at end of file
+add_test(NAME bls12-381 COMMAND unittests)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+  add_test(NAME bls12-381-nobmi2 COMMAND unittests)
+  set_tests_properties(bls12-381-nobmi2 PROPERTIES ENVIRONMENT "BLS_DISABLE_BMI2=1")
+endif()
```

Comment:
This of course means that if `ctest` is run on a CPU that does not have BMI2+ADX, the tests just silently run twice the non-BMI2+ADX code path. I'm reluctant to add much more complexity here. I'm thinking maybe Leap's AntelopeIO/leap#1132 is enough for now, since these tests will be run on leap's CI.